zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/ClusterConfigurationModifier.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.dynamic.config;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.dynamic.config.state.ClusterConfiguration;
+import io.camunda.zeebe.dynamic.config.state.DynamicPartitionConfig;
+import io.camunda.zeebe.dynamic.config.state.MemberState;
+import io.camunda.zeebe.dynamic.config.state.PartitionState;
+import io.camunda.zeebe.scheduler.ConcurrencyControl;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import java.util.Set;
+
+/**
+ * After the configuration is initialized, we can use a {@link ClusterConfigurationModifier} to
+ * update the initialized configuration. This process do not go through the usual process of adding
+ * a {@link io.camunda.zeebe.dynamic.config.state.ClusterConfigurationChangeOperation} to change the
+ * configuration. Instead, overwrites the configuration immediately after it is initialized. Hence,
+ * this should be used carefully.
```

Comment:
> Instead, overwrites the configuration immediately after it is initialized. Hence, this should be used carefully.

:thought_balloon: I wonder if there's anything we can do to avoid misuse in the future :sweat_smile:

Reply:
Depending on how this will get used in future, may be we can evolve the interface or add more pre-conditions. I don't have an idea otherwise.